packages/lesswrong/server/repos/VotesRepo.ts

```diff
@@ -283,20 +290,94 @@ class VotesRepo extends AbstractRepo<"Votes"> {
         }
       }
     }
-    
-    // On EAF, some reacts are anonymous (currently agree and disagree). For those, remove the userId.
-    if (isEAForum) {
-      reactionChanges = reactionChanges.map(change => {
-        if (eaAnonymousEmojiPalette.some(emoji => emoji.name === change.reactionType)) {
-          return {reactionType: change.reactionType}
-        }
-        return change
-      })
-    }
-    
+
     return reactionChanges;
   }
 
+  getEAKarmaChanges({
+    userId,
+    startDate,
+    endDate,
+    af,
+    showNegative = false,
+  }: KarmaChangesArgs): Promise<AnyKarmaChange[]> {
+    const powerField = af ? "afPower" : "power";
+    const {publicEmojis, privateEmojis} = getEAEmojisForKarmaChanges(showNegative);
+    const publicSelectors = publicEmojis.map((emoji) =>
+      `'${emoji}', ARRAY_AGG(
+        DISTINCT JSONB_BUILD_OBJECT('_id', v."userId", 'displayName', u."displayName")
+      ) FILTER (WHERE
+        v."cancelled" IS NOT TRUE AND
+        v."isUnvote" IS NOT TRUE AND
+        fm_vote_added_emoji(v."_id", '${emoji}')
```

Comment:
Does this add an extra query for every vote record targeting the current user? Will this be really bad for users who have high-profile posts but log in rarely?

Reply:
agree! Karma star seems like a good one to use for reacts too

Reply:
(as a response to your other comment, couldn't reply directly to that one)

Reply:
The sub-queries get inlined so the performance isn't bad until you start requesting multiple months (which always had bad performance anyway). I've added a limit so you can't request more than 40 days at once. I think this should be ~fine since it'll only affect users who have their batching frequency set to "realtime" (<100 users) and who don't login reqularly. Even when this happens, they probably only want to see the most recent karma changes anyway.

If performance ends up being bad then this should be fairly easy to cache since it's easy to detect when we need to invalidate.